Can be replaced with `misc-const-correctness.AllowedTypes` option of clang-tidy
that was added in https://github.com/llvm/llvm-project/issues/122592.
Needs a newer version of clang-tidy though.

--- contrib/restricted/googletest/googletest/include/gtest/internal/gtest-death-test-internal.h (index)
+++ contrib/restricted/googletest/googletest/include/gtest/internal/gtest-death-test-internal.h (working tree)
@@ -230,7 +230,7 @@ GTEST_API_ bool ExitedUnsuccessfully(int exit_status);
       goto GTEST_CONCAT_TOKEN_(gtest_label_, __LINE__);                        \
     }                                                                          \
     if (gtest_dt != nullptr) {                                                 \
-      std::unique_ptr< ::testing::internal::DeathTest> gtest_dt_ptr(gtest_dt); \
+      const std::unique_ptr< ::testing::internal::DeathTest> gtest_dt_ptr(gtest_dt); \
       switch (gtest_dt->AssumeRole()) {                                        \
         case ::testing::internal::DeathTest::OVERSEE_TEST:                     \
           if (!gtest_dt->Passed(predicate(gtest_dt->Wait()))) {                \



--- contrib/restricted/googletest/googletest/include/gtest/gtest-spi.h  (index)
+++ contrib/restricted/googletest/googletest/include/gtest/gtest-spi.h  (working tree)
@@ -153,10 +153,10 @@ GTEST_DISABLE_MSC_WARNINGS_POP_()  //  4251
       static void Execute() { statement; }                                    \
     };                                                                        \
     ::testing::TestPartResultArray gtest_failures;                            \
-    ::testing::internal::SingleFailureChecker gtest_checker(                  \
+    const ::testing::internal::SingleFailureChecker gtest_checker(            \
         &gtest_failures, ::testing::TestPartResult::kFatalFailure, (substr)); \
     {                                                                         \
-      ::testing::ScopedFakeTestPartResultReporter gtest_reporter(             \
+      const ::testing::ScopedFakeTestPartResultReporter gtest_reporter(       \
           ::testing::ScopedFakeTestPartResultReporter::                       \
               INTERCEPT_ONLY_CURRENT_THREAD,                                  \
           &gtest_failures);                                                   \
@@ -171,10 +171,10 @@ GTEST_DISABLE_MSC_WARNINGS_POP_()  //  4251
       static void Execute() { statement; }                                    \
     };                                                                        \
     ::testing::TestPartResultArray gtest_failures;                            \
-    ::testing::internal::SingleFailureChecker gtest_checker(                  \
+    const ::testing::internal::SingleFailureChecker gtest_checker(            \
         &gtest_failures, ::testing::TestPartResult::kFatalFailure, (substr)); \
     {                                                                         \
-      ::testing::ScopedFakeTestPartResultReporter gtest_reporter(             \
+      const ::testing::ScopedFakeTestPartResultReporter gtest_reporter(       \
           ::testing::ScopedFakeTestPartResultReporter::INTERCEPT_ALL_THREADS, \
           &gtest_failures);                                                   \
       GTestExpectFatalFailureHelper::Execute();                               \
@@ -217,11 +217,11 @@ GTEST_DISABLE_MSC_WARNINGS_POP_()  //  4251
 #define EXPECT_NONFATAL_FAILURE(statement, substr)                    \
   do {                                                                \
     ::testing::TestPartResultArray gtest_failures;                    \
-    ::testing::internal::SingleFailureChecker gtest_checker(          \
+    const ::testing::internal::SingleFailureChecker gtest_checker(    \
         &gtest_failures, ::testing::TestPartResult::kNonFatalFailure, \
         (substr));                                                    \
     {                                                                 \
-      ::testing::ScopedFakeTestPartResultReporter gtest_reporter(     \
+      const ::testing::ScopedFakeTestPartResultReporter gtest_reporter( \
           ::testing::ScopedFakeTestPartResultReporter::               \
               INTERCEPT_ONLY_CURRENT_THREAD,                          \
           &gtest_failures);                                           \
@@ -234,11 +234,11 @@ GTEST_DISABLE_MSC_WARNINGS_POP_()  //  4251
 #define EXPECT_NONFATAL_FAILURE_ON_ALL_THREADS(statement, substr)             \
   do {                                                                        \
     ::testing::TestPartResultArray gtest_failures;                            \
-    ::testing::internal::SingleFailureChecker gtest_checker(                  \
+    const ::testing::internal::SingleFailureChecker gtest_checker(            \
         &gtest_failures, ::testing::TestPartResult::kNonFatalFailure,         \
         (substr));                                                            \
     {                                                                         \
-      ::testing::ScopedFakeTestPartResultReporter gtest_reporter(             \
+      const ::testing::ScopedFakeTestPartResultReporter gtest_reporter(       \
           ::testing::ScopedFakeTestPartResultReporter::INTERCEPT_ALL_THREADS, \
           &gtest_failures);                                                   \
       if (::testing::internal::AlwaysTrue()) {                                \
